OpenSim/Common/Component.cpp

```diff
@@ -127,6 +127,24 @@ Component::Component(SimTK::Xml::Element& element) : Object(element)
 void Component::finalizeFromProperties()
 {
     reset();
+
+    // TODO use a flag to set whether we are lenient on having nameless
+    // Components. For backward compatibility we need to be able to 
+    // handle nameless components so assign them their class name
+    // - aseth
+    if (getName().empty()) {
+        setName(IO::Lowercase(getConcreteClassName()) + "_");
+    }
+
+    OPENSIM_THROW_IF( getName().empty(), ComponentHasNoName,
+                      getConcreteClassName() );
+
+    for (auto& comp : _memberSubcomponents) {
+        comp->setParent(*this);
+    }
+    for (auto& comp : _adoptedSubcomponents) {
+        comp->setParent(*this);
+    }
```

Comment:
Add comment that the property subcomponents get their parent set elsewhere. When we move to a unified iterator, it should not be used here.